src/app/result.ts

```diff
@@ -33,6 +37,7 @@ export class AsyncResult {
       if (timeout) {
         timeoutId = setTimeout(() => {
           clearInterval(intervalId);
+          this.status = "TIMEOUT";
```

Comment:
How about using `reject(error)` to represent "timeout" rather than just `resolve(null)`?
```suggestion
          reject(new Error("timeout"));
```
Then we can handle timeout without status.
```javascript
result.get(500).then(data => {
  console.log(data);
}).catch(e => {
  console.log(e.message);
});
```